### ISSUE TITLE
[Feature Request]: Rebirth Multiplayer Environment

### ISSUE THREAD

**Issue by foxfire667** [etternagame/etterna#1185]
### Is there an existing issue for the feature?

- [X] I have searched the existing feature requests

### Describe the Feature

Rebirth currently has no multiplayer, and the intention was at a later point to design a new multiplayer experience. After a large number of hours, this is the mockup and feature detail which encompasses how it could be integrated top to bottom. I hope that this will serve as a baseline which can be pulled from and built upon for future multiplayer development. 

This mockup collection and feature detail is simply too large to include as a GitHub issue, and can be read in its entirety on this document here: [Etterna Multiplayer Environment ](https://docs.google.com/document/d/1gfBnsIB6CpMcO2HmMqzmaWCAe5SDCeYyltEXxTY2paE)

From there you can see everything, and also download the mock-ups, PSDs, and icons used to further inspect or modify at your leisure. Additional eyes are welcome and discussion is encouraged. Just bear in mind that the document is massive, please look through it in whole if you intend to expand upon anything or try to poke holes. 

### How Does The Feature Add To The Game?

In short, it adds a seamless multiplayer experience to Rebirth. It is a full redesign which will allow for users to easily jump from single or multi play, in a manner which minimizes tedium, and is effectively entirely handled through a small chat window. This chat window also houses many other features, though some of them would require input from something like EO2. 

Ultimately, users can chat, direct message, add or remove friends, send invites, create or join rooms, and view a fair bit of user info for people that are online or share a multi room with them. There are also many command based settings, some new commands to ease multi use, spectator modes, multi default settings, and a new team battle setting to name a some big things this document covers.
